src/lib/reasoners/th_util.ml

```diff
@@ -30,16 +30,42 @@
 
 type answer = (Explanation.t * Expr.Set.t list) option
 
-
 type theory =
   | Th_arith
   | Th_sum
   | Th_adt
   | Th_arrays
   | Th_UF
+[@@deriving show]
+
+type limit_kind =
+  | Plus
+  | Minus
+  | None
```

Comment:
I agree that we shouldn't use `None`. The reason I use `Plus` or `Minus` is because I think the optimized value as a limit. In Math, we use to write `a+` for the limit from above and `a-` for the limit from below. I will use `Below | Above | Exact`.